src/components/OSCALComponentDefinition.js

```diff
@@ -17,24 +18,27 @@ const useStyles = makeStyles((theme) => ({
 export default function OSCALComponentDefinition(props) {
   const [error, setError] = useState(null);
   const [isLoaded, setIsLoaded] = useState(false);
+  const [inheritedProfilesAndCatalogs, setInheritedProfilesAndCatalogs] =
+    useState({});
   const classes = useStyles();
 
   useEffect(() => {
-    OSCALComponentResolveSources(
-      props.componentDefinition,
-      props.parentUrl,
-      () => {
-        setIsLoaded(true);
-        props.onResolutionComplete();
-      },
-      (errorReturned) => {
-        setError(errorReturned);
-        setIsLoaded(true);
-        props.onResolutionComplete();
-      }
+    setInheritedProfilesAndCatalogs(
+      OSCALComponentResolveSources(
+        props.componentDefinition,
+        props.parentUrl,
+        () => {
+          setIsLoaded(true);
+          props.onResolutionComplete();
+        },
+        (errorReturned) => {
+          setError(errorReturned);
+          setIsLoaded(true);
+          props.onResolutionComplete();
+        }
+      )
```

Comment:
This is getting a bit convoluted. Instead of having a method that takes in the return of the existing `OSCALComponentResolveSources` method, could we have an overall "resolve" wrapper method like the other OSCAL types do?